BRMS-319

adamconder opened the pull request.

adamconder committed:
Refactor packages and implementation

adamconder committed:
Refactor property names

adamconder committed:
Add comment

adamconder committed:
Refactor packages and switch check

adamconder committed:
BRMS-319: refactor Name parser into single package

adamconder committed:
BRMS-319: make Payload constructor arguments private

- Overriding getters for Payload constructors and now stripping whitespace from the string
- Removed method namesOriginalCase
- method names now does not lowercase string
- Now caching names in Names() object in Matching algorithm

adamconder committed:
BRMS-319: refactor MatchResult

- Refactored MatchResult class, now storing result of additionalNames match
- Now caching the Names() that we're matched on in the result object

adamconder committed:
BRMS-319: removing comments

adamconder committed:
BRMS-319: removing comment

adamconder committed:
BRMS-319: remove unused code

adamconder committed:
BRMS-319: add unit test for MatchingResult

adamconder committed:
BRMS-319: adding test for Names

Commit:
BRMS-319: Log additional name counts

Commit:
fixed tests.

Commit:
update log to info for coverage.

Commit:
BRMS-319: Refactor

Commit:
updated to fix tests.

Commit:
BRMS-319: Move record audit details out of TransactionAuditor

Commit:
updated to fix tests.

Commit:
BRMS-319: Move requestType check from commonutil into payload

Commit:
BRMS-319: Refactor TransactionAuditor

Commit:
BRMS-319: Refactor TestHelper

chris-ianson commented:
@adamconder @manishsw @anujakhare 

We've tidied up TransactionAuditor renaming and updating signature of methods. Record model includes audit method which builds audit for words/characters/flags (much clearer now)

Updated tests and refactored TestHelper a little. Moved request type check from CommonUtil into Payload

We really wanted to make the payload argument in audit required but it breaks WhereBirthRegisteredAudit when no Payload is available.

adamconder committed:
BRMS-319: Add interface BRMAPIDownstreamAudit

- removed duplicate methods
- refactored interfaces to now call transaction() instead of passing
as argument to audit()

adamconder committed:
Merge branch 'master' of github.com:hmrc/birth-registration-matching into BRMS-319

adamconder committed:
BRMS-319: add TODO for matching

adamconder committed:
BRMS-319: adding todo

Commit:
Added tests to ignore additional name when it is true.

Commit:
Merge branch 'BRMS-319' of github.com:hmrc/birth-registration-matching into BRMS-319

Commit:
updated tests for ignore additional name.

Commit:
BRMS-319: Add ignoreAdditionalNames in Payload

Commit:
BRMS-319: Remove additionalNames config value (now uses ignoreNiddleNames)

Commit:
BRMS-319: Move ignoreAdditionalNames check from Payload into FullMatching

Commit:
BRMS-319: Remove comments

Commit:
adding tests for ignoreAdditionalname -true scenario.

adamconder committed:
BRMS-319: add tests for Names object on response of matching

adamconder committed:
Merge branch 'BRMS-319' of github.com:hmrc/birth-registration-matching into BRMS-319

Commit:
BRMS-319: Merge branch 'BRMS-319' of github.com:hmrc/birth-registration-matching into BRMS-319 # Please enter a commit message to explain why this merge is necessary, # especially if it merges an updated upstream into a topic branch. # # Lines starting with '#' will be ignored, and an empty message aborts # the commit.

Commit:
BRMS-319: Merge branch 'BRMS-319' of github.com:hmrc/birth-registration-matching into BRMS-319 # Please enter a commit message to explain why this merge is necessary, # especially if it merges an updated upstream into a topic branch. # # Lines starting with '#' will be ignored, and an empty message aborts # the commit.

adamconder committed:
BRMS-319: removing comment

Commit:
updated for nameparser

Commit:
Merge branch 'BRMS-319' of github.com:hmrc/birth-registration-matching into BRMS-319

Commit:
BRMS-319: Merge branch 'BRMS-319' of github.com:hmrc/birth-registration-matching into BRMS-319 # Please enter a commit message to explain why this merge is necessary, # especially if it merges an updated upstream into a topic branch. # # Lines starting with '#' will be ignored, and an empty message aborts # the commit.

adamconder committed:
BRMS-319: removing comment

Commit:
BRMS-319: Merge branch 'BRMS-319' of github.com:hmrc/birth-registration-matching into BRMS-319 # Please enter a commit message to explain why this merge is necessary, # especially if it merges an updated upstream into a topic branch. # # Lines starting with '#' will be ignored, and an empty message aborts # the commit.

adamconder committed:
BRMS-319: removing comment

Commit:
Merge branch 'BRMS-319' of github.com:hmrc/birth-registration-matching into BRMS-319

Commit:
BRMS-319: Test ignoreAdditionalNames toggle in NameParserSpec

adamconder committed:
BRMS-319: removing comment

adamconder committed:
Merge branch 'BRMS-319' of github.com:hmrc/birth-registration-matching into BRMS-319

adamconder committed:
BRMS-319: FullMatching now uses Names for lastname
